### PR TITLE
Fix error-path memory leak in asn_mime.c

### PR DESCRIPTION
**Pull request by tmshort**
Need to free 'parts'.

### CODE REVIEW

File: crypto/asn1/asn_mime.c [hunk 1]
```diff
@@ -425,6 +425,8 @@ ASN1_VALUE *SMIME_read_ASN1(BIO *bio, BIO **bcont, const ASN1_ITEM *it)
             || hdr->value == NULL) {
             sk_MIME_HEADER_pop_free(headers, mime_hdr_free);
             ASN1err(ASN1_F_SMIME_READ_ASN1, ASN1_R_NO_SIG_CONTENT_TYPE);
+            sk_MIME_HEADER_pop_free(headers, mime_hdr_free);
```
Comment (bernd-edlinger) [inline crypto/asn1/asn_mime.c]: This is already done above.

Comment (tmshort) [reply crypto/asn1/asn_mime.c]: Yup, copy/paste error on my part... will update.

**Comment by mattcaswell** [comment]
Which branches does this apply to?

**Comment by tmshort** [comment]
> Which branches does this apply to?

Looks to be master, 1.1.0 and 1.0.2

**Comment by mattcaswell** [comment]
My approval counts for all branches. @bernd-edlinger?

**Comment by bernd-edlinger** [comment]
yes, good you asked....
mine counts for all branches as well.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**